backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -905,4 +911,90 @@ protected void redactSecretsForEndpoint(final SecurityContext securityContext, f
+            internalDeleteEventTypeFromEndpoint(securityContext, eventTypeId, endpointId);
+        } else {
+            legacyRBACDeleteEventTypeFromEndpoint(securityContext, eventTypeId, endpointId);
+        }
+    }
+
+    @RolesAllowed(ConsoleIdentityProvider.RBAC_WRITE_NOTIFICATIONS)
+    protected void legacyRBACDeleteEventTypeFromEndpoint(final SecurityContext securityContext, final UUID eventTypeId, final UUID endpointId) {
+        internalDeleteEventTypeFromEndpoint(securityContext, eventTypeId, endpointId);
+    }
+
+    private void internalDeleteEventTypeFromEndpoint(final SecurityContext securityContext, final UUID eventTypeId, final UUID endpointId) {
+        final String orgId = getOrgId(securityContext);
+        endpointEventTypeRepository.deleteEndpointFromEventType(eventTypeId, endpointId, orgId);
+    }
+
+    @PUT
+    @Path("/{endpointId}/eventType/{eventTypeId}")
+    @Operation(summary = "Add a link between an endpoint and an event type", description = "Add a link between an endpoint and an event type.")
+    @RolesAllowed(ConsoleIdentityProvider.RBAC_WRITE_NOTIFICATIONS)
+    @APIResponses(value = {
+        @APIResponse(responseCode = "204", content = @Content(mediaType = APPLICATION_JSON, schema = @Schema(type = SchemaType.STRING))),
+        @APIResponse(responseCode = "404", content = @Content(mediaType = TEXT_PLAIN,  schema = @Schema(type = SchemaType.STRING)),
+            description = "No event type or endpoint found with the passed id.")
+    })
+    @Tag(name = OApiFilter.PRIVATE)
+    public void addEventTypeToEndpoint(@Context final SecurityContext securityContext, @RestPath final UUID eventTypeId, @RestPath final UUID endpointId) {
+        if (this.backendConfig.isKesselRelationsEnabled()) {
+            this.kesselAuthorization.hasPermissionOnResource(securityContext, IntegrationPermission.EDIT, ResourceType.INTEGRATION, endpointId.toString());
+
+            internalAddEventTypeToEndpoint(securityContext, eventTypeId, endpointId);
+        } else {
+            legacyRbacAddEventTypeToEndpoint(securityContext, eventTypeId, endpointId);
+        }
+    }
+
+    public void legacyRbacAddEventTypeToEndpoint(final SecurityContext securityContext, final UUID eventTypeId, final UUID endpointId) {
+        internalAddEventTypeToEndpoint(securityContext, eventTypeId, endpointId);
+    }
+
+    private void internalAddEventTypeToEndpoint(final SecurityContext securityContext, final UUID eventTypeId, final UUID endpointId) {
+        final String orgId = getOrgId(securityContext);
+        endpointEventTypeRepository.addEventTypeToEndpoint(eventTypeId, endpointId, orgId);
+    }
+
+    @PUT
+    @Path("/{endpointId}/eventTypes")
+    @Operation(summary = "Update  links between an endpoint and event types", description = "Update  links between an endpoint and event types.")
+    @RolesAllowed(ConsoleIdentityProvider.RBAC_WRITE_NOTIFICATIONS)
```

Comment:
This annotation needs to be moved to `legacyRbacUpdateEventTypesLinkedToEndpoint`.